packages/components/src/font-size-picker/test/index.tsx

```diff
@@ -1,290 +1,430 @@
-					'XL'
-				);
+		it( 'calls onChange when a font size is selected', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
 			} );
-			it( 'should use font size `slug` for for header hint label by default', () => {
-				const fontSizes = [
-					{
-						name: 'Allosaurus Large',
-						slug: 'allosaurus-l',
-						size: '20rem',
-					},
-				];
-				render(
-					<FontSizePicker
-						fontSizes={ fontSizes }
-						value={ fontSizes[ 0 ].size }
-						__nextHasNoMarginBottom
-					/>
-				);
+			const onChange = jest.fn();
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					onChange={ onChange }
+				/>
+			);
+			await user.click( screen.getByRole( 'radio', { name: 'Medium' } ) );
+			expect( onChange ).toHaveBeenCalledWith( '16px' );
+		} );
 
-				const largeFontSizeElement = screen.getByLabelText(
-					'Size Allosaurus Large(rem)'
-				);
-				expect( largeFontSizeElement ).toBeInTheDocument();
+		commonTests( fontSizes );
+	} );
+
+	function commonTests( fontSizes: FontSize[] ) {
+		it( 'allows custom values by default', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
 			} );
-			it( 'should fallback to font size `slug` for header hint label if `name` is undefined', () => {
-				const fontSizes = [
-					{
-						slug: 'gigantosaurus',
-						size: '1000px',
-					},
-				];
-				render(
-					<FontSizePicker
-						fontSizes={ fontSizes }
-						value={ fontSizes[ 0 ].size }
-						__nextHasNoMarginBottom
-					/>
-				);
+			const onChange = jest.fn();
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					onChange={ onChange }
+				/>
+			);
+			await user.click(
+				screen.getByRole( 'button', { name: 'Set custom size' } )
+			);
+			await user.type( screen.getByLabelText( 'Custom' ), '80' );
+			expect( onChange ).toHaveBeenCalledWith( '80px' );
+		} );
 
-				const giganticFontSizeElement = screen.getByLabelText(
-					'Size gigantosaurus(px)'
-				);
-				expect( giganticFontSizeElement ).toBeInTheDocument();
+		it( 'does not allow custom values when disableCustomFontSizes is set', () => {
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					disableCustomFontSizes
+				/>
+			);
+			expect(
+				screen.queryByRole( 'button', { name: 'Set custom size' } )
+			).not.toBeInTheDocument();
+		} );
+
+		it( 'does not display a slider by default', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
 			} );
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+				/>
+			);
+			await user.click(
+				screen.getByRole( 'button', { name: 'Set custom size' } )
+			);
+			// TODO: There are two spinbuttons with similar names, so can't
+			// select the slider's input in an accessible way.
```

Comment:
Isn't there only 1 spinbutton (the UnitControl to the side of the slider) ? The slider itself should have role="slider" if I remember correctly?

(This applies also to other tests performing the same query)

Reply:
They don't have `role="slider"` (does such a role exist?) but it is labelled. Not sure why I thought it wasn't. Fixed!